Sources/Target/Common/ThreadBase.cpp

```diff
@@ -30,5 +30,17 @@ ErrorCode ThreadBase::modifyRegisters(
   action(state);
   return writeCPUState(state);
 }
+
+ErrorCode ThreadBase::beforeResume() {
+  if (!_process->isAlive())
+    return kErrorProcessNotFound;
+
+  BreakpointManager *bpm = _process->hardwareBreakpointManager();
+  if (bpm != nullptr) {
+    bpm->enable();
```

Comment:
This seems odd to me. If I understand correctly, this will enable hardware watchpoints on *all* threads. Wouldn't it be better to call `HardwareBreakpointManager::enable(..., Thread::Thread *thread)` and pass `this` as argument so we enable them only on the current thread that was newly created?